OpenRA.Mods.Common/Activities/UndeployForGrantedCondition.cs

```diff
@@ -24,8 +24,26 @@ public UndeployForGrantedCondition(Actor self, GrantConditionOnDeploy deploy)
 			this.deploy = deploy;
 		}
 
+		protected override void OnFirstRun(Actor self)
+		{
+			if (self.Info.HasTraitInfo<TurretedInfo>())
+			{
+				var tInfo = self.Info.TraitInfo<TurretedInfo>();
```

Comment:
Please use `.TraitInfoOrDefault` and a null check instead of `.HasTraitInfo` followed by `.TraitInfo`.

Reply:
@abcdefg30 
Applied the changes.